process/track/blockProcessor_test.go

```diff
@@ -1,3 +1,743 @@
+	headersIndexes := make([]int, 0)
+	sortedHeaders := []data.HeaderHandler{&block2.Header{Nonce: 1}}
+	bp.GetNextHeader(&longestChainHeadersIndexes, headersIndexes, nil, sortedHeaders, 0)
+
+	assert.Equal(t, 0, len(longestChainHeadersIndexes))
+}
+
+func TestGetNextHeader_ShouldReturnEmptySliceWhenSortedHeadersHaveHigherNonces(t *testing.T) {
+	t.Parallel()
+	blockProcessorArguments := CreateBlockProcessorMockArguments()
+	bp, _ := track.NewBlockProcessor(blockProcessorArguments)
+
+	longestChainHeadersIndexes := make([]int, 0)
+	headersIndexes := make([]int, 0)
+	prevHeader := &block2.Header{}
+	sortedHeaders := []data.HeaderHandler{&block2.Header{Nonce: 2}}
+	bp.GetNextHeader(&longestChainHeadersIndexes, headersIndexes, prevHeader, sortedHeaders, 0)
+
+	assert.Equal(t, 0, len(longestChainHeadersIndexes))
+}
+
+func TestGetNextHeader_ShouldReturnEmptySliceWhenHeaderConstructionIsNotValid(t *testing.T) {
+	t.Parallel()
+	blockProcessorArguments := CreateBlockProcessorMockArguments()
+	bp, _ := track.NewBlockProcessor(blockProcessorArguments)
+
+	longestChainHeadersIndexes := make([]int, 0)
+	headersIndexes := make([]int, 0)
+	prevHeader := &block2.Header{}
+	sortedHeaders := []data.HeaderHandler{&block2.Header{Nonce: 1}}
+	bp.GetNextHeader(&longestChainHeadersIndexes, headersIndexes, prevHeader, sortedHeaders, 0)
+
+	assert.Equal(t, 0, len(longestChainHeadersIndexes))
+}
+
+func TestGetNextHeader_ShouldReturnEmptySliceWhenHeaderFinalityIsNotChecked(t *testing.T) {
+	t.Parallel()
+	hasherMock := &mock.HasherMock{}
+	marshalizerMock := &mock.MarshalizerMock{}
+
+	blockProcessorArguments := CreateBlockProcessorMockArguments()
+	bp, _ := track.NewBlockProcessor(blockProcessorArguments)
+
+	longestChainHeadersIndexes := make([]int, 0)
+	headersIndexes := make([]int, 0)
+
+	header1 := &block2.Header{
+		Round: 1,
+		Nonce: 1,
+	}
+	header1Marshalized, _ := marshalizerMock.Marshal(header1)
+	headerHash1 := hasherMock.Compute(string(header1Marshalized))
+
+	header2 := &block2.Header{
+		Round:    2,
+		Nonce:    2,
+		PrevHash: headerHash1,
+	}
+
+	sortedHeaders := []data.HeaderHandler{header2}
+	bp.GetNextHeader(&longestChainHeadersIndexes, headersIndexes, header1, sortedHeaders, 0)
+
+	assert.Equal(t, 0, len(longestChainHeadersIndexes))
+}
+
+func TestGetNextHeader_ShouldWork(t *testing.T) {
+	t.Parallel()
+	hasherMock := &mock.HasherMock{}
+	marshalizerMock := &mock.MarshalizerMock{}
+
+	blockProcessorArguments := CreateBlockProcessorMockArguments()
+	bp, _ := track.NewBlockProcessor(blockProcessorArguments)
+
+	longestChainHeadersIndexes := make([]int, 0)
+	headersIndexes := make([]int, 0)
+
+	header1 := &block2.Header{
+		Round: 1,
+		Nonce: 1,
+	}
+	header1Marshalized, _ := marshalizerMock.Marshal(header1)
+	headerHash1 := hasherMock.Compute(string(header1Marshalized))
+
+	header2 := &block2.Header{
+		Round:    2,
+		Nonce:    2,
+		PrevHash: headerHash1,
+	}
+	header2Marshalized, _ := marshalizerMock.Marshal(header2)
```

Comment:
core.CalculateHash? valid for the rest of the file

Reply:
Done